src/components/TopView.tsx

```diff
@@ -1,95 +1,315 @@
-  hiddenTab: {
-    display: 'none !important'
-  }
+    hiddenTab: {
+        display: "none !important"
+    }
 }));
 
 function TabPanels(props: TabPanelsProps) {
-  const styles = tabPanelsUseStyles();
-  return (
-    <div className={props.className}>
-      {props.children.map((child, index) => (
-        <div
-          key={index}
-          className={index === props.value ? '' : styles.hiddenTab}
-        >
-          {child}
+    const styles = tabPanelsUseStyles();
+    return (
+        <div className={props.className}>
+            {props.children.map((child, index) => (
+                <div
+                    key={index}
+                    className={index === props.value ? "" : styles.hiddenTab}
+                >
+                    {child}
+                </div>
+            ))}
         </div>
-      ))}
-    </div>
-  );
+    );
 }
 
 const useStyles = makeStyles((theme) => ({
-  topView: {
-    display: 'flex',
-    flexDirection: 'column'
-  },
-  paper: {
-    flex: 1,
-    minHeight: 0
-  },
-  tabPanel: {
-    flex: 1,
-    display: 'flex',
-    flexDirection: 'column',
-    minHeight: 0,
-    '& > *': {
-      flex: 1,
-      display: 'flex',
-      flexDirection: 'column',
-      minHeight: 0,
-      overflowY: "auto",
-      '& > *': {
+    topView: {
+        display: "flex",
+        flexDirection: "column"
+    },
+    paper: {
         flex: 1,
         minHeight: 0
-      }
+    },
+    tabPanel: {
+        flex: 1,
+        display: "flex",
+        flexDirection: "column",
+        minHeight: 0,
+        "& > *": {
+            flex: 1,
+            display: "flex",
+            flexDirection: "column",
+            minHeight: 0,
+            overflowY: "auto",
+            "& > *": {
+                flex: 1,
+                minHeight: 0
+            }
+        }
     }
-  }
 }));
 
 export function TopView() {
-  const [value, setValue] = useState(0);
-  const handleChange = (event: any, newValue: number) => {
-    setValue(newValue);
-  };
-  const styles = useStyles();
-
-  return (
-    <div className={styles.topView}>
-      <Paper className={styles.topView + ' ' + styles.paper}>
-        <TabContext value={String(value)}>
-          <Tabs textColor="primary" value={value} onChange={handleChange}>
-            <Tab label="Data Visualization" />
-            <Tab label="File Upload" />
-            <Tab label="Tutorials" />
-          </Tabs>
-          <TabPanels className={styles.tabPanel} value={value}>
-            <DataView />
-            <Paper variant="outlined">
-              <FileUpload />
+    const [value, setValue] = useState(0);
+    const [datasets, setDatasets] = useState([]);
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const handleChange = (event: any, newValue: number) => {
+        setValue(newValue);
+    };
+
+    const [user] = useAuthState(firebase.auth());
+
+    const styles = useStyles();
+
+    useEffect(() => {
```

Comment:
I feel like this thing would be better in a different file (maybe `FileUpload.tsx`). TopView is only supposed to hold the individual tabs, I don't think we should start adding logic into it.